aten/src/ATen/mps/MPSStream.mm

```diff
@@ -112,7 +119,13 @@ @interface MPSGraphExecutionDescriptor ()
 void MPSStream::flush() {
   if (_commandBuffer) {
     [_commandBuffer commit];
-    [_commandBuffer release];
+    // if commitAndContinue is disabled (e.g., for Profiler), we keep the command
```

Comment:
As you suggested, I put and assert to assign `prevCommandBuffer` if only profiler is enabled. Without the assert the logic of using `prevCommandBuffer` makes sense too, because we encode nothing on it and we nullify `_commandBuffer` and create a new one and encode on the new one always.